Sources/PackageModel/Module.swift

```diff
@@ -129,9 +130,24 @@ public class ClangModule: CModule {
     }
 }
 
+/// A dictionary containing mapping of C family source extension to their file type in Xcode.
+private let xcodeCLanguageFileTypeDict = [
+    ".c": "sourcecode.c.c",
+    ".m": "sourcecode.c.objc",
+    ".cxx": "sourcecode.cpp.cpp",
+    ".cc": "sourcecode.cpp.cpp",
+    ".cpp": "sourcecode.cpp.cpp",
+    ".mm": "sourcecode.cpp.objcpp",
+]
+
 extension ClangModule: XcodeModuleProtocol {
-    public var fileType: String {
-        return "sourcecode.c.c"
+    public func fileType(forSource source: String) -> String {
```

Comment:
I thought about it but then extracting file extension or iterating in dict would be similar complexity I think?